dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,519 @@
+from __future__ import unicode_literals
+
+from time import sleep
+import posixpath
+import os
+import logging
+
+try:
+    import google_auth_oauthlib
+    from .oauth2 import OAuth2
+except ImportError:
+    google_auth_oauthlib = None
+
+from requests import ConnectionError
+
+from dvc.scheme import Schemes
+from dvc.path.gdrive import PathGDrive
+from dvc.utils import tmp_fname, move
+from dvc.utils.compat import urlparse, makedirs
+from dvc.remote.base import RemoteBASE
+from dvc.config import Config
+from dvc.progress import progress
+from dvc.remote.gdrive.utils import track_progress, only_once, get_chunk_size
+
+
+logger = logging.getLogger(__name__)
+
+
+class GDriveError(Exception):
+    pass
+
+
+class GDriveResourceNotFound(GDriveError):
+    def __init__(self, path):
+        super(GDriveResourceNotFound, self).__init__(
+            "'{}' resource not found".format(path)
+        )
+
+
+class RemoteGDrive(RemoteBASE):
+    """Google Drive remote implementation
+
+    Example URLs:
+
+    Datasets/my-dataset inside "My Drive" folder:
+
+        gdrive://root/Datasets/my-dataset
+
+    Folder by ID (recommended):
+
+        gdrive://1r3UbnmS5B4-7YZPZmyqJuCxLVps1mASC
+
+        (get it https://drive.google.com/drive/folders/{here})
+
+    Dataset named "my-dataset" in the hidden application folder:
+
+        gdrive://appDataFolder/my-dataset
+
+        (this one wouldn't be visible through Google Drive web UI and
+         couldn't be shared)
+    """
+
+    scheme = Schemes.GDRIVE
+    REGEX = r"^gdrive://.*$"
+    REQUIRES = {"google-auth-oauthlib": google_auth_oauthlib}
+    PARAM_CHECKSUM = "md5Checksum"
+    GOOGLEAPIS_BASE_URL = "https://www.googleapis.com/"
+    MIME_GOOGLE_APPS_FOLDER = "application/vnd.google-apps.folder"
+    SPACE_DRIVE = "drive"
+    SCOPE_DRIVE = "https://www.googleapis.com/auth/drive"
+    SPACE_APPDATA = "appDataFolder"
+    SCOPE_APPDATA = "https://www.googleapis.com/auth/drive.appdata"
+    TIMEOUT = (5, 60)
+
+    # Default credential is needed to show the string of "Data Version
+    # Control" in OAuth dialog application name and icon in authorized
+    # applications list in Google account security settings. Also, the
+    # quota usage is limited by the application defined by client_id.
+    # The good practice would be to suggest the user to create their
+    # own application credentials.
+    DEFAULT_CREDENTIALPATH = os.path.join(
+        os.path.dirname(__file__), "google-dvc-client-id.json"
+    )
+
+    def __init__(self, repo, config):
+
+        super(RemoteGDrive, self).__init__(repo, config)
```

Comment:
minor: can we keep the same style for `__init__` everywhere? sometimes you put extra spaces, sometimes not, sometimes very aggressively :) I'm not sure what the pattern is :)

Reply:
@shcheklein I've seen it in other patches from other guys. Looks like editor (e.g. pycharm) is doing that sometimes. See no harm, and it doesn't seem to violate pep8.

Reply:
yep, thus minor ... it's not Pycharm though (unless there is some setting for it I don't know about).